### PR TITLE
curl: Updated to 8.6.0

### PR DESCRIPTION
**Pull request by aholovskyi**
Update `curl` to 8.6.0 (pull from https://github.com/openwrt/packages/tree/openwrt-22.03/net/curl )

References: OWF-2806

### CODE REVIEW

File: net/curl/Config.in [hunk 1]
```diff
@@ -139,7 +149,7 @@ config LIBCURL_ZSTD
 
 config LIBCURL_UNIX_SOCKETS
 	bool "Enable unix domain socket support"
-	default n
+	default y
```
Comment (s-skorobreshchuk) [inline net/curl/Config.in:152]: Why this change is needed? Maybe worse to leave small explanation in description for historical needs...

Comment (aholovskyi) [reply net/curl/Config.in:152]: It is actually not needed

**Review by s-skorobreshchuk — APPROVED** [review]
